tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -137,6 +137,66 @@ def fabs_phrase_r_i(x : 'int', y : 'float'):
     assert(isclose(f2(x,-y), fabs_phrase_r_i(x,-y), rtol=RTOL, atol=ATOL))
     assert(isclose(f2(-x,y), fabs_phrase_r_i(-x,y), rtol=RTOL, atol=ATOL))
 
+#------------------------------ isnan function ----------------------------#
+def test_numpy_isnan(language):
+    def numpy_isnan_test(x : 'float'):
+        from numpy import isnan
+        return isnan(x)
+
+    f = epyccel(numpy_isnan_test, language=language)
+
+    input_data = np.nan
+    expected_output = numpy_isnan_test(input_data)
+    
+    assert f(input_data) == expected_output
+    assert matching_types(f(input_data), expected_output)
+
+    input_data = np.random.uniform(-1e6, 1e6)
+    expected_output = numpy_isnan_test(input_data)
+    
+    assert f(input_data) == expected_output
+    assert matching_types(f(input_data), expected_output)
+
+#------------------------------ isinf function ----------------------------#
+def test_numpy_isinf(language):
+    def numpy_isinf_test(x : 'float'):
+        from numpy import isinf
+        return isinf(x)
+
+    f = epyccel(numpy_isinf_test, language=language)
+
+    input_data = np.inf
+    expected_output = numpy_isinf_test(input_data)
+
+    assert f(input_data) == expected_output
+    assert matching_types(f(input_data), expected_output)
+
+    input_data = np.random.uniform(-1e6, 1e6)
+    expected_output = numpy_isinf_test(input_data)
+    
+    assert f(input_data) == expected_output
+    assert matching_types(f(input_data), expected_output)
+
+#------------------------------ isfinite function ----------------------------#
+def test_numpy_isfinite(language):
+    def numpy_isfinite_test(x : 'float'):
+        from numpy import isfinite
+        return isfinite(x)
+
+    f = epyccel(numpy_isfinite_test, language=language)
+
+    input_data = np.random.uniform(-1e6, 1e6)
+    expected_output = numpy_isfinite_test(input_data)
+    
+    assert f(input_data) == expected_output
+    assert matching_types(f(input_data), expected_output)
+
+    input_data = np.inf
+    expected_output = numpy_isfinite_test(input_data)
+    
+    assert f(input_data) == expected_output
+    assert matching_types(f(input_data), expected_output)
```

Comment:
It would be great to add some tests that take array arguments. If the `is_elemental` property of the `PyccelInternalFunction` is set then these tests should work